### PR TITLE
Run tests by Ruby 2.3.3 and ruby-head

### PR DESCRIPTION
**Pull request by ojiry**
#### Purpose

Test with latest ruby version.
Ruby 2.4 will be released at an early date, so I want to add ruby-head.

#### Changes

Modify ruby version of .travis.yml.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,7 +6,8 @@ rvm:
   - 2.0
   - 2.1
   - 2.2
-  - 2.3.1
+  - 2.3.3
+  - ruby-head
```
Comment (koic) [inline .travis.yml]: `ruby-head` is under development that is not stable. I think that add `ruby-head` to `allow_failures`.

cf. https://github.com/octokit/octokit.rb/blob/3887fd52134f56032b871304e56bd0f1753b0dc5/.travis.yml#L27-L30

**Comment by ojiry** [comment]
Thanks @koic ! You’re right.

File: .travis.yml [hunk 2]
```diff
@@ -28,6 +28,7 @@ matrix:
   allow_failures:
     - rvm: rbx-2
     - rvm: jruby
+    - rvm: ruby-head
```
Comment (koic) [inline .travis.yml]: `ruby-head` is also required in the`rvm` section. `ruby-head` is not running as follows.

https://travis-ci.org/octokit/octokit.rb/builds/181598215

Comment (ojiry) [inline .travis.yml]: Oops!
Let me sleep on it.

**Comment by ojiry** [comment]
I updated rspec version because Travis CI does not work.
https://github.com/octokit/octokit.rb/pull/844/commits/93219e1f10f6363f2616e35a4c4eb2c5ba8a2574

File: .travis.yml (hunk 1 above)
Comment (koic) [inline .travis.yml]: This looks good.

**Pull request closed without merging**